runtime/src/asset.rs

```diff
@@ -1197,6 +1206,96 @@ decl_module! {
             Self::deposit_event(RawEvent::IdentifiersUpdated(ticker, identifiers));
             Ok(())
         }
+
+        /// Whitelisting the Smart-Extension address for a given ticker
+        ///
+        /// # Arguments
+        /// * `origin` - Signer who owns to ticker/asset
+        /// * `ticker` - ticker for whom extension get added
+        /// * `extension_details` - Details of the smart extension
+        pub fn add_extension(origin, ticker: Ticker, extension_details: SmartExtension<T::AccountId>) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+            let sender_key = Key::try_from(sender.encode())?;
+            let my_did =  match <identity::Module<T>>::current_did() {
```

Comment:
It makes sense to make this block a separate function since it is copied two other times.

Reply:
This is something already noted in another task. So this cleanup will be done at that time.